### PR TITLE
Adding test cases for label functions and fix existing test warnings

### PR DESCRIPTION
**Pull request by vedhav**
Partially addresses #82 

Changes made:
1. Added test cases to cover the `xportr_df_label()` from the **`df_label.R`** file
2. Added test cases to cover the `xportr_label()` from the **`label.R`** file
3. Wrapped `as.list()` around a map function to suppress deprecation warning from purrr
4. Fixed the commented test by removing the regex filter. So, now it works as expected.
5. Added `metacore` package as Suggests because we are starting to use it in tests, and soon we will use it in the examples.

I have not yet added the metacore package to the `renv.lock` Do we even need a renv.lock for this package? I feel just the DESCRIPTION file should be enough WDYT?

### CODE REVIEW

File: tests/testthat/test-metadata.R [hunk 1]
```diff
@@ -8,34 +11,110 @@ extract_format <- function(.x) {
...
-                                   row.names = c(NA, -1L), label = "Label"))
+  df_spec_labeled_df <- xportr_df_label(df, df_meta)
+  piped_spec_labeled_df <- df %>% xportr_df_label(df_meta)
+  renamed_spec_labeled_df <- xportr_df_label(renamed_dataset_df, metacore_meta)
+  domain_spec_labeled_df <- xportr_df_label(renamed_dataset_df, metacore_meta, domain = "DOMAINNAME")
+
+  expect_equal(attr(df_spec_labeled_df, "label"), "Label") 
+  expect_equal(
+    dput(df_spec_labeled_df),
+    structure(list(x = "a", y = "b"), class = "data.frame", row.names = c(NA, -1L), label = "Label")
```
Comment (bms63) [inline tests/testthat/test-metadata.R:92]: could we add some more line breaks here and maybe even break these tests into smaller chunks if possible?

Comment (vedhav) [reply tests/testthat/test-metadata.R:92]: Yeah, that makes sense. I will split them into smaller chunks. I also like your suggestion to have better test_that descriptions and the convention followed by the admiral is great and it's worth doing that kind of `function_name Test n: SINGLE TEST CASE` descriptions. I will create a task to do this across all our tests.

Comment (vedhav) [reply tests/testthat/test-metadata.R:92]: Created issue #92 to make sure we standardize the test description across all of our tests. However, I was unable to add it to the board. Maybe you can add it there?

File: DESCRIPTION [hunk 2]
```diff
@@ -50,6 +50,7 @@ Roxygen: list(markdown = TRUE)
 RoxygenNote: 7.2.3
 Suggests: 
     testthat (>= 3.0.0),
+    metacore,
```
Comment (elimillera) [inline DESCRIPTION:53]: I don't think we should make a dependency on metacore. If metacore ever gets pulled off we run the risk of xportr getting pulled off as well.

**Review by elimillera — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 2 above)
Comment (vedhav) [reply DESCRIPTION]: That makes sense, I will remove it.

**Comment by bms63** [comment]
skip_if `metacore` is not available on CRAN.

File: tests/testthat/test-metadata.R [hunk 3]
```diff
@@ -7,122 +6,518 @@ extract_format <- function(.x) {
   format_
 }
 
-test_that("Variable label", {
+extract_var_label <- function(.x) {
+  vapply(.x, function(.x) attr(.x, "label"), character(1), USE.NAMES = FALSE)
+}
+
+test_that("xportr_label: Correctly applies label for data.frame spec", {
```
Comment (bms63) [inline tests/testthat/test-metadata.R:13]:
```suggestion
test_that("xportr_label: Correctly applies label from data.frame spec", {
```

File: tests/testthat/test-metadata.R [hunk 4]
```diff
@@ -7,122 +6,518 @@ extract_format <- function(.x) {
...
+        y = structure("b", label = "bar")
+      ),
+      row.names = c(NA, -1L),
+      `_xportr.df_arg_` = "DOMAIN",
+      class = "data.frame"
+    )
+  )
+})
+
+test_that("xportr_label: Correctly applies label for metacore spec", {
```
Comment (bms63) [inline tests/testthat/test-metadata.R:75]:
```suggestion
test_that("xportr_label: Correctly applies label from metacore spec", {
```

File: tests/testthat/test-metadata.R [hunk 5]
```diff
@@ -7,122 +6,518 @@ extract_format <- function(.x) {
...
+  df_meta <- data.frame(
+    dataset = "df",
+    label = strrep("a", 41)
+  )
+
+  expect_error(
+    xportr_df_label(df, df_meta),
+    "dataset label must be 40 characters or less"
+  )
+})
```
Comment (bms63) [inline tests/testthat/test-metadata.R:233]: I think these are duplicates of the above tests?

Comment (vedhav) [reply tests/testthat/test-metadata.R:233]: As discussed on the call, Pretty much the only thing that changed is the function we're testing.

**Review by bms63 — APPROVED** [review]
Thank you!!

**Pull request merged**